### ISSUE TITLE
Reverse index vs. inverted index

### ISSUE THREAD

**Issue by dharmaturtle** [donnemartin/system-design-primer#390]
Hi!

[Design a key-value cache to save the results of the most recent web server queries](https://github.com/donnemartin/system-design-primer/blob/master/solutions/system_design/query_cache/README.md#design-a-key-value-cache-to-save-the-results-of-the-most-recent-web-server-queries) and [Design a web crawler](https://github.com/donnemartin/system-design-primer/blob/master/solutions/system_design/web_crawler/README.md) both are described as using a "Reverse index service". I believe this should be an _inverted_ index. Reversing the index simply reverses the key, e.g. `123` becomes `321` and speeds up inserts (at the expense of slowing down range queries. [Wiki](https://en.wikipedia.org/wiki/Reverse_index), [StackOverflow](https://stackoverflow.com/questions/19882785/what-is-the-point-of-reverse-indexing). An [inverted index](https://en.wikipedia.org/wiki/Inverted_index) helps with searching, which I believe is the point of the above two architectures.

I may be entirely wrong - pretty new to all this!

**Comment by bshaibu** [comment]
@dharmaturtle I'm pretty sure you're right - even the original [Google paper](http://infolab.stanford.edu/~backrub/google.html) refers to these sort of indexes as an _inverted_ index. Reverse indexes are very different.

I'd open a PR, but I'm not sure how to update the images.

**Comment by yzhong52** [comment]
@donnemartin I agree that they should be called "inverted index". This is a bit misleading to call them "reverse index" here. Can you advice how we can have this fixed?
